meson.build

```diff
@@ -5,21 +5,76 @@
 # license: MIT
 #
 project('unity', 'c',
-    license: 'MIT',
-    meson_version: '>=0.37.0',
-    default_options: ['werror=true', 'c_std=c11'])
+  license: 'MIT',
+
+  # Set project version to value extracted from unity.h header
+  version: run_command(
+    [
+      find_program('python', 'python3'),
+      'auto' / 'extract_version.py',
+      'src' / 'unity.h'
+    ],
+    check: true
+  ).stdout().strip(),
+
+  # `meson.project_source_root()` introduced in 0.56.0
+  meson_version: '>=0.56.0',
+  default_options: [
+    'werror=true',
+    'c_std=c11'
+  ]
+)
+
+build_fixture = get_option('extension_fixture').enabled()
+build_memory = get_option('extension_memory').enabled()
```

Comment:
You're just using these like booleans, it seems misleading to allow people to specify either `=auto` or `=disabled` and have auto really just mean disabled.

What about defining the options as type boolean, and just checking the option value directly?